8Knot/queries/pr_response_query.py

```diff
@@ -0,0 +1,137 @@
+import logging
+import pandas as pd
+from db_manager.augur_manager import AugurManager
+from app import celery_app
+from cache_manager.cache_manager import CacheManager as cm
+import io
+import datetime as dt
+from sqlalchemy.exc import SQLAlchemyError
+
+QUERY_NAME = "PR_RESPONSE"
+
+
+@celery_app.task(
+    bind=True,
+    autoretry_for=(Exception,),
+    exponential_backoff=2,
+    retry_kwargs={"max_retries": 5},
+    retry_jitter=True,
+)
+def pr_response_query(self, repos):
+    """
+    (Worker Query)
+    Executes SQL query against Augur database for contributor data.
+
+    This query gets the time to first response on a pr if it exists,
+    if not the earliest_msg_timestamp is null
+
+    Args:
+    -----
+        repo_ids ([str]): repos that SQL query is executed on.
+
+    Returns:
+    --------
+        dict: Results from SQL query, interpreted from pd.to_dict('records')
+    """
+    logging.warning(f"{QUERY_NAME}_DATA_QUERY - START")
+
+    if len(repos) == 0:
+        return None
+
+    query_string = f"""
```

Comment:
Not for sure what you mean by "roles" of the tables used. Adding more context in in general. TBH to me the table names seems pretty self descriptive

Reply:
They are, generally, but reading english is easier than SQL.

Stating "Gathers all of the messages on a PR, including those that directly respond to the PR and those that are in Reviews, and finds the earliest" is nicer to read, and then easier to apply to the SQL.

Reply:
You raise a good point- I do agree that it'd be messy to document Augur stuff in queries. The best Augur documentation I can find for tables is either here: https://github.com/chaoss/augur/tree/main/docs/source/schema or by manually going into DBeaver and hovering over tables. The Augur schema is relatively informative, but the natural language contents of the tables isn't always available. This would be a really useful documentation effort that would be work the Augur team's time.

Reply:
@JamesKunstle Augur team is already in process of updating those docs. Ill make an issue for us to link it in our docs once its complete

Reply:
https://github.com/oss-aspen/8Knot/issues/511

Reply:
++